### ISSUE TITLE
VirtualField does not work with `count: true` and match property the same time

### ISSUE THREAD

**Issue by JustDoItSascha** [Automattic/mongoose#11714]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
When you want to use a virtualField in combination with a match query and `count: true`, than the match query is not used.

**If the current behavior is a bug, please provide the steps to reproduce.**
```typescript
ExampleSchema.virtual('users', {
    ref: 'User',
    localField: 'userId',
    foreignField: '_id',
    count: true,
    match: {active: true},
});
```

**What is the expected behavior?**
When I have two users (one active, the other not) then the virtual field `users` should contain `1`. But I get `2`. I have taken a look into the mongo debug logs and I can see that the query has not been sent to mongo. It's filtered out in mongoose.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Mongoose: 6.2.2

**Comment by IslandRhythms** [comment]
Can't say if this bug is the same/related to the one you're experiencing but I've been able to reproduce where the match filter is being ignored. Please modify this below script to demonstrate your bug if this doesn't also reproduce yours.

```javascript

const mongoose = require('mongoose');

const AuthorSchema = new mongoose.Schema({
    name: String
});

const BlogPostSchema = new mongoose.Schema({
    title: String,
    author: { type: mongoose.Schema.Types.ObjectId, ref: 'Author' },
    comments: [{
      author: { type: mongoose.Schema.Types.ObjectId, ref: 'Author' },
      content: String
    }]
  });

AuthorSchema.virtual('posts', {
    ref: 'BlogPost',
    localField: '_id',
    foreignField: 'author',
    count: true,
    match: { name: 'Test Testerson' } // match option with basic query selector
  });
  
  const Author = mongoose.model('Author', AuthorSchema, 'Author');
  const BlogPost = mongoose.model('BlogPost', BlogPostSchema, 'BlogPost');

async function run() {
    await mongoose.connect('mongodb://localhost:27017');
    await mongoose.connection.dropDatabase();

    const entry = await Author.create({ name: 'Test Testerson' });
    const otherEntry = await Author.create({ name: 'Testing Testerson'});
    const blogEntry = await BlogPost.create({ title: 'Moby Dick', author: entry._id, comments: [{ author: entry._id, content: 'Moby Dick is an old book' }]});
    const otherBlogEntry = await BlogPost.create({ title: 'Cathcer in the Rye', author: otherEntry._id, comments: [{ author: otherEntry._id, content: 'Required Reading in High School'}]})
      // After population
    const author = await Author.find().populate('posts');
  
    console.log(author) // Array of not `archived` posts
}

run();
```

**Comment by JustDoItSascha** [comment]
There are some mistakes in the script, but it's nearly the same:

```
const mongoose = require('mongoose');

const AuthorSchema = new mongoose.Schema({
    name: String
});

const BlogPostSchema = new mongoose.Schema({
    title: String,
    author: String,
  });

AuthorSchema.virtual('posts', {
    ref: 'BlogPost',
    localField: '_id',
    foreignField: 'author',
    count: true,
    match: { name: 'Test Testerson' } // match option with basic query selector
  });
  
  const Author = mongoose.model('Author', AuthorSchema, 'Author');
  const BlogPost = mongoose.model('BlogPost', BlogPostSchema, 'BlogPost');

async function run() {
    await mongoose.connect('mongodb://localhost:27017');
    await mongoose.connection.dropDatabase();

    const entry = await Author.create({ name: 'Test Testerson' });
    const otherEntry = await Author.create({ name: 'Testing Testerson'});
    const blogEntry = await BlogPost.create({ title: 'Moby Dick', author: entry.name});
    const otherBlogEntry = await BlogPost.create({ title: 'Cathcer in the Rye', author: otherEntry.name})
      // After population
    const author = await Author.findOne({ name: 'Test Testerson' }).populate('posts');
  
    console.log(author.posts) // Shows 2 instead of 1
}

run();
```

**Comment by vkarpov15** [comment]
@JustDoItSascha your script doesn't print "2", it prints "0" because there's no blog post with name "Test Testerson". Also because you're referencing authors by name, not `_id`. The below script is more correct, and demonstrates that `count` does in fact work with `match`. Can you please modify the below script to demonstrate your issue?

```javascript
const mongoose = require('mongoose');

const AuthorSchema = new mongoose.Schema({
    name: String
});

const BlogPostSchema = new mongoose.Schema({
    title: String,
    author: 'ObjectId',
  });

mongoose.set('debug', true);

AuthorSchema.virtual('posts', {
    ref: 'BlogPost',
    localField: '_id',
    foreignField: 'author',
    count: true,
    match: { title: 'Moby Dick' } // match option with basic query selector
  });
  
  const Author = mongoose.model('Author', AuthorSchema, 'Author');
  const BlogPost = mongoose.model('BlogPost', BlogPostSchema, 'BlogPost');

async function run() {
    await mongoose.connect('mongodb://localhost:27017/test');
    await mongoose.connection.dropDatabase();

    const entry = await Author.create({ name: 'Test Testerson' });
    const otherEntry = await Author.create({ name: 'Testing Testerson'});
    const blogEntry = await BlogPost.create({ title: 'Moby Dick', author: entry._id});
    const otherBlogEntry = await BlogPost.create({ title: 'Catcher in the Rye', author: entry._id});
      // After population
    const author = await Author.findOne({ name: 'Test Testerson' }).populate('posts');
  
    console.log(author.posts) // Shows 2 instead of 1
}

run();
```

**Comment by JustDoItSascha** [comment]
@vkarpov15 I have to apologize, you are right! And there is also no bug. It was a bug on my side, because I used a field for matching which was not present in the schema.

We are using microservices and the schema in one microservice was not updated and the field was missing there. But we don't use it elsewhere in this microservice, so we did not recognize it. I'm sorry! Your code helped me to think of that, so thank you very much!
